### DISCUSSION TITLE
Allow login to crates.io using arbitrary auth providers

### DISCUSSION THREAD

**Discussion by ghost** [rust-lang/crates.io#6149] [Q&A]
With each update to GitHub delving more and more into AI and shady {business,privacy,etc} practices since the acquisition by Microsoft, I am looking to remove myself from this particular ecosystem as soon as possible. It makes me very sad that a language and community that I've grown to love is still this inflexible when it comes to being able to authenticate.

Since #326 is locked, what is the desired method for discussing the dethroning of GitHub as the sole auth provider for `crates.io`?

Apologies if this has been asked before. I did try searching for existing discussions, but frankly the discussions UI is a joke.

**Comment by Turbo87** [comment]
there are currently several factors that make it hard for us to allow different auth providers:

- crates.io usernames currently match the GitHub usernames, so if we allow a different provider we will need a way to avoid conflicts
- the current system for assigning teams as crate owners is unfortunately directly tied to GitHub teams
- from what I was told, GitHub is somewhat good at preventing spam accounts and other malicious activity, which we can't handle ourselves with the existing resources of the project

there might be other factors I've forgotten, but those are the three main ones that I can remember from the last discussion.

the way forward would be to find solutions for all of them, writing an RFC to propose these solutions and then implement the necessary changes. for the second point we have a rough idea in mind already, but the others are still tbd. you're welcome to help out.

**Reply by carols10cents** [reply]
Additionally, the code and database are quite tied to the assumption that GitHub is the only login (the users table has column `gh_login` and `gh_id` for example), so significant refactorings need to take place to enable adding other services. No one has had time to do this work yet.

#326 is locked because we agree that we want to allow non-GitHub accounts; we don't need any further discussion/convincing on that point. What we need now is people to do the work.

**Reply by carols10cents** [reply]
I've updated the issue description at #326 with what we've discussed here.
